ipynb/primary-processing/11-generate_per_bodysite_results.md

```diff
@@ -0,0 +1,278 @@
+...                     agu.get_existing_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['source-%s-weighted-unifrac' % depth]),
+...                 ]
+...
+...                 sink_dir = agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-dir' % depth])
+...                 sink_otu = agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-otu' % depth])
+...                 sink_map = agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-map' % depth])
+...                 sink_beta = [
+...                     agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-unweighted-unifrac' % depth]),
+...                     agu.get_path(agenv.paths['package']['all_participants_all_samples'][bodysite][trim]['sink-%s-weighted-unifrac' % depth]),
+...                     ]
+...
+...                 yield (bodysite, depth, source_otu, source_map, source_alpha, source_beta, sink_dir, sink_otu, sink_map, sink_beta)
+```
+
+Let's use our helper function to add the alpha diversity to our mapping file, and move the rest of the data we need.
+
+```python
+>>> for (bodysite, depth, source_otu, source_map, source_alpha, source_beta, sink_dir, sink_otu, sink_map, sink_beta) in rarefied_parameter_iterator():
+...     # Creates the sink directory
+...     !mkdir -p $sink_dir
+...
+...     # Loads the mapping file
+...     map_ = pd.read_csv(source_map, sep='\t', dtype=str)
+...     map_.set_index('#SampleID', inplace=True)
+...
+...     # Loads the alpha diveristy file
+...     alphas = {metric : pd.read_csv(fp, sep='\t', dtype=str).set_index('Unnamed: 0').transpose()
+...               for (metric, fp) in source_alpha.iteritems()
+...              }
+...
+...     # Adds alpha diversity to the map
+...     alpha_map = agu.add_alpha_diversity(map_, alphas)
+...     alpha_map.to_csv(sink_map, sep='\t', index_label='#SampleID')
+...
+...     # Moves the OTU table
+...     !cp $source_otu $sink_otu
+...
+...     # Moves the beta diversity tables
+...     for source, sink in zip(*(source_beta, sink_beta)):
+...         !cp $source $sink
+```
+
+We're going to select a single sample for each subject at each body site. The human microbiome project [doi: ] demonstrated that differences across bodysites are larger than interpersonal differences within a single bodysite. However, other studies (ie Caporaso 2011 [[doi: 10.1186/gb-2011-12-5-r50](http://www.ncbi.nlm.nih.gov/pubmed/21624126)], Wu 2011 [[doi: 10.1126/science](http://www.ncbi.nlm.nih.gov/pubmed/21885731)]) suggest a high degree of stability within individual communities. A single sample can be use to avoid the bias associated with multiple closely related samples from a single individual.
```

Comment:
This is also an outdated change. Again, not sure why Git isn't showing the new version.